pandas/_libs/window_indexer.pyx

```diff
@@ -0,0 +1,167 @@
+# cython: boundscheck=False, wraparound=False, cdivision=True
+
```

Comment:
I think resaonable to make pandas/_libs/window/indexerx.pyx (and move window.pyx here); can be a pre or post PR.